File: 769.最多能完成排序的块.cpp
/*
 * @lc app=leetcode.cn id=769 lang=cpp
 *
 * [769] 最多能完成排序的块
 */

// @lc code=start
class Solution {
public:
    int maxChunksToSorted(vector<int> &arr)
    {
        int ans = 0, curMax = 0;
        for (int i = 0; i < arr.size(); ++i)
        {
            curMax = max(curMax, arr[i]);
            if (curMax == i)
                ++ans;
        }
        return ans;
    }
};
// @lc code=end

File: 769.最多能完成排序的块.py
#
# @lc app=leetcode.cn id=769 lang=python3
#
# [769] 最多能完成排序的块
#

# @lc code=start
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ans = 0
        pmax = -1
        for i, x in enumerate(arr):
            pmax = max(pmax, x)
            if pmax == i:
                ans += 1
        return ans
# @lc code=end

